### PR TITLE
feat: enable use of GEOJson external layers

### PR DESCRIPTION
**Pull request by jenniferarnesen**
Implements: https://dhis2.atlassian.net/browse/DHIS2-15981

If geojson external overlay layers are configured for instance, they can now be added to a map.
The data table will display the layer data as long as the collection is of homogenous geo types.
User can also click on a feature and the right panel will open, showing the feature data.

### CODE REVIEW

File: src/loaders/geoJsonUrlLoader.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        featureStyle = layer.featureStyle || EMPTY_FEATURE_STYLE
+    }
+
+    let geoJson
+    let error
+
+    try {
+        geoJson = await fetchData(newConfig.url, engine, baseUrl)
+    } catch (error) {
+        console.error(error)
```
Comment (turban) [inline src/loaders/geoJsonUrlLoader.js:78]: Remove console if we show the error in the layer card?

File: src/util/geojson.js [hunk 2]
```diff
@@ -115,3 +115,90 @@ export const getCoordinatesBounds = (coordinates) =>
             [Number.NEGATIVE_INFINITY, Number.NEGATIVE_INFINITY],
         ]
     )
+
+const TYPE_NUMBER = 'number'
+const TYPE_STRING = 'string'
+// const TYPE_DATE = 'date'
```
Comment (turban) [inline src/util/geojson.js:121]: Remove?

**Review by turban — APPROVED** [review]
Looks good! Approved with minor comments. Still needs to be monkey tested!

File: src/components/datatable/useTableData.js [hunk 3]
```diff
@@ -147,13 +175,18 @@ const getEarthEngineHeaders = ({ aggregationType, legend, data }) => {
...
+    getGeojsonDisplayData(firstDataItem)
+
 const EMPTY_AGGREGATIONS = {}
 const EMPTY_LAYER = {}
 
 export const useTableData = ({ layer, sortField, sortDirection }) => {
     const allAggregations = useSelector((state) => state.aggregations)
     const aggregations = allAggregations[layer.id] || EMPTY_AGGREGATIONS
 
+    const errorCode = useRef(null)
```
[the remark this answers is not in the document]
Comment (jenniferarnesen) [reply src/components/datatable/useTableData.js:188]: I'll look into it during soft freeze 💯

**Pull request merged**